docs/compilers/CSharp/Runtime Async Design.md

```diff
@@ -27,20 +27,32 @@ We use the following helper APIs to indicate suspension points to the runtime, i
 ```cs
 namespace System.Runtime.CompilerServices;
 
-// These methods are used to await things that cannot use runtime async signature form
 // TODO: Clarify which of these should be preferred? Should we always emit the `Unsafe` version when awaiting something that implements `ICriticalNotifyCompletion`?
 namespace System.Runtime.CompilerServices;
 
 public static class RuntimeHelpers
 {
-    [RuntimeAsyncMethod]
+    // These methods are used to await things that cannot use the Await helpers below
+    [MethodImpl(MethodImplOptions.Async)]
     public static Task AwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : INotifyCompletion;
-    [RuntimeAsyncMethod]
+    [MethodImpl(MethodImplOptions.Async)]
     public static Task UnsafeAwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : ICriticalNotifyCompletion;
+
+    // These methods are used to directly await method calls
+    // TODO: Do we only use these when directly awaiting a method call, or do we always use these when a `Task` is awaited, even if that task is a local?
+    [MethodImpl(MethodImplOptions.Async)]
+    public static void Await(Task task);
+    [MethodImpl(MethodImplOptions.Async)]
+    public static T Await<T>(Task<T> task);
+    [MethodImpl(MethodImplOptions.Async)]
+    public static void Await(ValueTask task);
+    [MethodImpl(MethodImplOptions.Async)]
+    public static T Await<T>(ValueTask<T> task);
 }
 ```
 
-We presume the following `MethodImplOptions` bit is present. This is used to indicate to the JIT that it should generate an async state machine for the method.
+We presume the following `MethodImplOptions` bit is present. This is used to indicate to the JIT that it should generate an async state machine for the method. This bit is not allowed to be used manually on any method; it is added by the compiler
+to an `async` method.
```

Comment:
Do you want to add a note for the rule Andy proposed yesterday (method marked with MethodImpl.Async and not returning a Task may not be invoked in user code)?